SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -366,3 +378,304 @@ void CIncNSSolver::BC_ConjugateHeat_Interface(CGeometry *geometry, CSolver **sol
+     
+          su2double VelNormal = GeometryToolbox::DotProduct(int(MAXNDIM), Vel, UnitNormal);
+
+          su2double VelTang[MAXNDIM] = {0.0};
+          for (auto iDim = 0u; iDim < nDim; iDim++)
+            VelTang[iDim] = Vel[iDim] - VelNormal*UnitNormal[iDim];
+
+          su2double VelTangMod = GeometryToolbox::Norm(int(MAXNDIM), VelTang);
+
+          /*--- Compute normal distance of the interior point from the wall ---*/
+
+          su2double WallDist[MAXNDIM] = {0.0};
+          for (auto iDim = 0u; iDim < nDim; iDim++)
+            WallDist[iDim] = (Coord[iDim] - Coord_Normal[iDim]);
+
+          WallDistMod = 0.0;
+          for (auto iDim = 0u; iDim < nDim; iDim++)
+            WallDistMod += WallDist[iDim]*WallDist[iDim];
+          WallDistMod = sqrt(WallDistMod);
+
+          /*--- Compute mach number ---*/
+
+          // M_Normal = VelTangMod / sqrt(Gamma * Gas_Constant * T_Normal);
+
+          /*--- Compute the wall temperature using the Crocco-Buseman equation ---*/
+
+          //T_Normal = T_Wall * (1.0 + 0.5*Gamma_Minus_One*Recovery*u_normal*u_normal);
+          // this means that T_Wall = T_Normal/(1.0+0.5*Gamma_Minus_One*Recovery*u_normal*u_normal)
+          //T_Wall = T_Normal/(1.0+0.5*Gamma_Minus_One*Recovery*VelTangMod*VelTangMod);
+          // in incompressible flows, we can assume that there is no velocity-related temperature change
+          // Prandtl: T+ = Pr*y+
+          T_Wall = nodes->GetTemperature(iPoint); 
+
+          /*--- Extrapolate the pressure from the interior & compute the
+           wall density using the equation of state ---*/
+
+          // we assume no additional influence on the normal point
+          P_Wall = nodes->GetPressure(iPoint);
+          Density_Wall = nodes->GetDensity(iPoint);
+          Lam_Visc_Wall = nodes->GetLaminarViscosity(iPoint);
+          su2double Conductivity_Wall = nodes->GetThermalConductivity(iPoint);
+          su2double Density_Normal = nodes->GetDensity(Point_Normal);
+          su2double Lam_Visc_Normal = nodes->GetLaminarViscosity(Point_Normal);
+
+          /*--- Compute the shear stress at the wall in the regular fashion
+           by using the stress tensor on the surface ---*/
+
+          grad_primvar  = nodes->GetGradient_Primitive(iPoint);
+
+          su2double div_vel = 0.0;
+          for (auto iDim = 0u; iDim < nDim; iDim++)
+            div_vel += grad_primvar[iDim+1][iDim];
+
+          for (auto iDim = 0u; iDim < nDim; iDim++) {
+            for (auto jDim = 0u ; jDim < nDim; jDim++) {
+              Delta = 0.0; if (iDim == jDim) Delta = 1.0;
+              tau[iDim][jDim] = Lam_Visc_Wall*(  grad_primvar[jDim+1][iDim]
+                                               + grad_primvar[iDim+1][jDim]) -
+              TWO3*Lam_Visc_Wall*div_vel*Delta;
+            }
+            for (auto jDim = 0u; jDim < nDim; jDim++)
+              TauElem[iDim] += tau[iDim][jDim]*UnitNormal[jDim];
+          }
```

Comment:
There is a method in CNumerics to compute the stress tensor that can be used directly.

Reply:
oh yes, computestresstensor, I totally missed this/forgot about this!